lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -46,34 +47,101 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
                 filter_func=possibly_scored,
                 start_node=self.location,
         ):
-            descendant = course_structure[descendant_key]
+            self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
-            if not getattr(descendant, 'has_score', False):
-                continue
+        self.all_total, self.graded_total = graders.aggregate_scores(self.scores, self.display_name, self.location)
 
+    def save(self, student, subsection, course):
+        """
+        Persist the SubsectionGrade.
+        """
+        visible_blocks = [
+            BlockRecord(unicode(location), weight, score.possible)
```

Comment:
`BlockRecord` is just a named tuple, so it doesn't do any special handling to serialize a `UsageKey`. I figured serializing to unicode here was the right move, since we're preparing to save the data (not use it in memory any further).

Without any serialization, we get this error:
```
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/new/subsection_grade.py", line 72, in save
    visible_blocks=visible_blocks,
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py", line 242, in save_grade
    grade, is_created = cls.objects.get_or_create(user_id=user_id, usage_key=usage_key, defaults=kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/models/query.py", line 407, in get_or_create
    return self._create_object_from_params(lookup, params)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/models/query.py", line 439, in _create_object_from_params
    obj = self.create(**params)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py", line 168, in create
    visible_blocks_hash = VisibleBlocks.objects.hash_from_blockrecords(blocks=visible_blocks)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py", line 107, in hash_from_blockrecords
    model = self.create_from_blockrecords(blocks)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py", line 94, in create_from_blockrecords
    model, _ = self.get_or_create(hashed=blocks.to_hash(), defaults={'blocks_json': blocks.to_json()})
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py", line 75, in to_hash
    self._hash = b64encode(sha1(self.to_json()).digest())
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py", line 52, in to_json
    sort_keys=True,
  File "/usr/lib/python2.7/json/__init__.py", line 250, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
  File "/usr/lib/python2.7/json/encoder.py", line 431, in _iterencode
    for chunk in _iterencode_list(o, _current_indent_level):
  File "/usr/lib/python2.7/json/encoder.py", line 332, in _iterencode_list
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 442, in _iterencode
    o = _default(o)
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: BlockUsageLocator(CourseLocator(u'edX', u'DemoX', u'Demo_Course', None, None), u'problem', u'303034da25524878a2e66fb57c91cf85') is not JSON serializable
```